### ISSUE TITLE
Missing window button icons from the GTK+ application built using vcpkg environment 

### ISSUE THREAD

**Issue by kakyoism** [microsoft/vcpkg#7827]
Although I successfully built the GNOME official example using vcpkg, the running application main windows shows empty maximize/minimize/close buttons, with console output:
```bash
(hello_gtk.exe:1936): Gtk-WARNING **: Could not find the icon 'window-minimize-symbolic-ltr'. The 'hicolor' theme
was not found either, perhaps you need to install it.
You can get a copy from:
        http://icon-theme.freedesktop.org/releases
```

From a casual stackoverflow search, I noticed that most similar questions are about MSYS2, where installing an icon-set package or fix the relative folder structure would solve the issue. But with vcpkg, I found that certain magic happens when building the app: All the dependency DLLs will be copied to the EXE folder. This makes it hard to understand what's going on with the relative paths. 

Any ideas how to fix this?

**Comment by kakyoism** [comment]
I worked around the problem by fixing the icons manually. [The GTK doc](https://www.gtk.org/download/windows.php) was a big help.

However, having that 3-paragraph instruction for a manual hack is a shame. There should be a one-click fix in vcpkg, even if there are legal concerns.

**Comment by Neumann-A** [comment]
> All the dependency DLLs will be copied to the EXE folder. This makes it hard to understand what's going on with the relative paths.

That has to do with where Windows searches for DLLs. You can toogle that feature by VCPKG_APPLOCAL_DEPS. If you deactivate it, dont forget to add the triplets/(debug/)bin folder to PATH variable or you will be greated with an error message when starting your application

> I worked around the problem by fixing the icons manually.

did you need to install anything extra on top of gtk or did you just move folders around?

**Comment by kakyoism** [comment]
> did you need to install anything extra on top of gtk or did you just move folders around?

Like I mentioned in my last message, I basically did what GTK doc says: Click that inline link and jump to Section `the Building and distributing your application`.

**Comment by Neumann-A** [comment]
hmm at least the old windows style should work correctly which could be considered a vcpkg issue. 
Installing other/additional themes or icons should be done by your cmake script and is not a vcpkg issue.

**Comment by kakyoism** [comment]
> hmm at least the old windows style should work correctly which could be considered a vcpkg issue.
> Installing other/additional themes or icons should be done by your cmake script and is not a vcpkg issue.

Agreed. The default Windows theme should be available out of the box.

**Comment by napoleonlijun** [comment]
Have you fix this?

**Comment by sam20908** [comment]
Looking forward for this to be fixed too. I had to redistribute a copy of the entire "share" folder in MSYS2 when installing the icons in there. It isn't convenient to redistribute such a big folder along with my application.

**Comment by jrm86** [comment]
Having the same issue. The "GTK doc" linked above is a dead link

**Comment by sam20908** [comment]
It'd be cool if vcpkg can use their downloaded MSYS2 to download the icons and treat them "like DLLs" so they get copied to the right destination.

**Comment by kakyoism** [comment]
> Having the same issue. The "GTK doc" linked above is a dead link.
> 
> Edit: Found the page, despite the dead link. [Here it is](https://www.gtk.org/docs/installations/windows/). Some of the instructions there are really not specific enough. Is the stuff in Step 2 and Step 3 going into the `share/themes/Windows10/gtk-3.0` directory, or do they get their own directory? This isn't clear by the statement "Perform the same steps" in Step 3. Where is `share/glib-2.0/schemas` coming from in Step 5? Should it already exist, or am I creating it? What is its contents?
> 
> In short, there's a lot of unstated assumptions being made and details left out of the instructions. For someone having zero experience with GTK, I'm left with a brute force "keep moving stuff around in the directories until it works" kind of approach.
> 
> I know I'm mostly commenting on the GTK documentation, but their poor documentation should be a good motivation to make installation easier here for Windows (or Microsoft can write a better alternative documentation for this issue).

Completely agreed. These are not hard problems but probably need traction to get the dev attention.

**Comment by Kristopher38** [comment]
I'm having a very similar issue, has anyone figured a fix for missing icons? Fixing the relative folder structure didn't help at all, I tried a few configurations but no luck, and as said previously the docs on this are awful. The only way I found to make it work is to run the application placed inside MSYS2 environment (i.e. your MSYS2 home directory) from the MSYS2 console itself, but I want to get rid of those requirements.

**Comment by procedural** [comment]
Here's a Windows 10-themed 32-bit GTK program I wrote that you can use to see how and if the styling works for you: https://github.com/procedural/magicacsg_copy_layers_program

**Comment by az0** [comment]
FWIW, I am successfully using icons in GTK3 3.24. I'm having sort of the opposite problem: unused icons are bloating my application, and while trying to debloat my app's bundled GTK3, I learned how GTK icons are stored.

- `gtk-3-vs17.dll` contains a GTK GResource bundle with Adwaita and HighContrast theme resources: 468 PNG icons, 6 SVG icons, 41 .ui files, 10 .css files, and 1 emoji .data file
- I build 32-bit GTK and Python for Windows in the [pygtkwin repo](https://github.com/bleachbit/pygtkwin)
- [This PowerShell script](https://github.com/bleachbit/bleachbit/blob/master/windows/python-gtk3-install.ps1) setups up my Python + GTK environment
- I use Adwaita by default but also switch to a [bundled Windows 10 theme](https://github.com/bleachbit/bleachbit/tree/master/themes/windows10)
- Users can switch in real time (no restart required) between both dark and light variants of the Adwaita and Windows 10 themes.
